src/apps/statements/expressPresenter/utils/validateStatementProcessingPriority.ts

```diff
@@ -0,0 +1,17 @@
+import { createWarning, Warnings } from 'rulr';
+
+import { StatementProcessingPriority } from '../../enums/statementProcessingPriority.enum';
+
+export const validateStatementProcessingPriority = (
+  statementProcessingPriority: string | undefined,
+): void | Warnings[] => {
+  if (
+    statementProcessingPriority &&
+    !Object.values(StatementProcessingPriority).includes(
+      statementProcessingPriority as StatementProcessingPriority,
+    )
+  ) {
+    const warnings = [createWarning(statementProcessingPriority, ['query', 'priority'])];
+    throw new Warnings({}, ['query'], warnings);
+  }
+};
```

Comment:
Wow nicely done! At some point we should switch the xapi-service and xapi-validation to [Rulr v8](https://github.com/ryansmith94/rulr) because it's much easier to use and create your own rules.

Reply:
@ryansmith94 Yes, I also wanted a fresh version of rulr when was doing this 😆